src/clients/uploads-api-client.ts

```diff
@@ -0,0 +1,56 @@
+import { AmazonMarketplace } from '@scaleleap/amazon-marketplaces'
+import globalAxios, { AxiosInstance } from 'axios'
+
+import {
+  Configuration,
+  ConfigurationParameters,
+  UploadsApi,
+  UploadsApiCreateUploadDestinationForResourceRequest,
+} from '../api-models/uploads-api-model'
+import { BASE_PATH } from '../api-models/uploads-api-model/base'
+
+export type UploadsApiClientCreateUploadDestinationForResourceRequest = Omit<
+  UploadsApiCreateUploadDestinationForResourceRequest,
+  'marketplaceIds'
+>
+
+export interface UploadsApiClientConfigurationParameters
+  extends Omit<ConfigurationParameters, 'basePath'> {
+  marketplace?: AmazonMarketplace
+}
+
+export class UploadsApiClient {
+  private uploadsApi: UploadsApi
+
+  constructor(
+    protected configuration?: UploadsApiClientConfigurationParameters,
+    protected axios: AxiosInstance = globalAxios,
+  ) {
+    // TODO: assert marketplace
+    const basePath: string | undefined = configuration?.marketplace?.sellingPartner?.region.endpoint
+    const uploadsApiConfiguration: Configuration = new Configuration({
+      ...configuration,
+      basePath,
+    })
+    this.uploadsApi = new UploadsApi(uploadsApiConfiguration, BASE_PATH, axios)
+  }
+
+  public createUploadDestinationForResource(
+    apiClientParameters: UploadsApiClientCreateUploadDestinationForResourceRequest,
+    options?: any,
+  ) {
+    const marketplaceId = this.configuration?.marketplace?.id
+    let marketplaceIds: string[] = []
+    if (marketplaceId) {
+      marketplaceIds = [marketplaceId]
```

Comment:
@moltar : The docs say: [Max count is 1.](https://github.com/amzn/selling-partner-api-docs/blob/main/references/uploads-api/uploads_2020-11-01.md#user-content-parameters:~:text=Max%20count%20%3A%201)
🤔 

I will check on Production API. 😄

Reply:
Ha, you are right. But on some other calls you can supply more than one. Maybe they just wanted to keep the interface design. Or maybe they are just bad at API design 🤷🏼

Reply:
But that is almost irrelevant, because we do not want to extend too much anyways. Just keep things as simple as possible and as close as possible to the original API models.